keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/DeliveryConfigRepositoryTests.kt

```diff
@@ -532,6 +534,26 @@ abstract class DeliveryConfigRepositoryTests<T : DeliveryConfigRepository, R : R
               expectThat(pausedRepository.getPause(PauseScope.RESOURCE, resource.id)).isNull()
             }
           }
+
+          test("deletes any older versions of resources") {
+            with(deliveryConfig.resources.first()) {
+              val newResourceVersion = resource(
+                kind = kind,
+                id = spec.id,
+                application = application
+              )
+              resourceRepository.store(newResourceVersion)
+            }
+
+            expectThat(repository.get(deliveryConfig.name).resources)
+              .hasSize(4)
+              .map { it.version }
+              .containsExactlyInAnyOrder(1, 1, 1, 2)
+
+            repository.deleteByApplication(deliveryConfig.application)
+
+            expectThat(resourceRepository.getResourcesByApplication(deliveryConfig.application)).isEmpty()
```

Comment:
Delete the delivery config and check there are now zero resources associated with that application. This failed before because the version 1 of the updated resource would not get deleted